pkg/user/tx_client.go

```diff
@@ -132,6 +133,16 @@ func WithEstimatorService(conn *grpc.ClientConn) Option {
 	}
 }
 
+// WithAdditionalCoreEndpoints adds additional core endpoints to the TxClient.
+// For transaction submission, the client will attempt to use the primary endpoint
+// and the first two additional endpoints provided via this option.
+func WithAdditionalCoreEndpoints(conns []*grpc.ClientConn) Option {
+	// set multiple core endpoints on the `TxClient`
+	return func(c *TxClient) {
+		c.conns = append(c.conns, conns...)
```

Comment:
IMO here, you should already trim conn set so that len(conns) <= 3 

log.Debug a warning that says "provided %d additional core endpoints, can only use 2"